File: week5/matrix-diagonal-sum.cpp
class Solution {
public:
    int diagonalSum(vector<vector<int>>& mat) {
        int i=0,j=0,sum=0;
        while(i<mat.size() && j<mat.size()){
            sum+=mat[i][j];
            i++;
            j++;
        }
        i=mat.size()-1;
        j=0;
        while(j<mat.size() and i>=0){
            sum+=mat[i][j];
            j++;
            i--;
        }
        if(mat.size()%2==1)
        sum-=mat[mat.size()/2][mat.size()/2];
        return sum;
    }
};

File: week5/matrix-diagonal-sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        sum=0
        i=0
        j=0
        while i<len(mat) and j<len(mat):
            sum+=mat[i][j]
            i+=1
            j+=1
        i=len(mat)-1
        j=0
        while i>=0 and j<len(mat):
            sum+=mat[i][j]
            i-=1
            j+=1
        if len(mat)%2==1:
            sum-=mat[len(mat)//2][len(mat)//2]
        return sum
        